Sources/SwiftGenKit/Stencil/ContextHelpers.swift

```diff
@@ -0,0 +1,21 @@
+//
+// SwiftGen
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+
+// MARK: - Lazy evaluation
+
+// We need to use a `NSObject` subclass + `@objc` to make Stencil use KVC to access this property,
+// and thus allow that `lazy var` not to be resolved by Stencil too early or before it's accessed in the template
+final class StencilContextLazyDocument: NSObject {
```

Comment:
At first I thought having a protocol here, something like `DictionarySerializable` or something like that, with a `toDictionary()` method.

But then the types such as LazyDocument here would have to implement that protocol in SGK, which shouldn't happen. Test details should **not** be in normal source code.